### PR TITLE
Fix #3889 : remove piped stream for file download

### LINKED ISSUE #3889: Copying big files is slow (compared to kubectl)
### Is your task related to a problem? Please describe

On a local network, `kubectl cp` is network bounded when copying files : 
```bash
# Copying a 800MB file
$ time kubectl cp pod_name:/path/to/big/file ./big_file
tar: Removing leading `/' from member names
 
real    0m7.548s
user    0m2.456s
sys     0m3.299s
```
Using the java kube client, time is much higher. It is cpu bounded : 
```java
  public static void main(String[] args) {
    DefaultKubernetesClient kubeClient = new DefaultKubernetesClient();
    final PodResource<Pod> pod = kubeClient.pods().withName("pod_name");
    pod.file("/path/to/big/file").copy(Paths.get("./big_file")); // takes 77s (10x !)
  }
```
I'm running java11 but it does not seem to be the problem. After a first analysis, it comes from Piped(In/Out)putStream that makes heavy use of locks.
Removing use of pipes drops the download time from ~77s to ~11s (7x gain).

### Describe the solution you'd like

Refactor the code to make the network receiving thread directly write in the file instead of using piped(in/out)putstreams.

### Describe alternatives you've considered

I've considered forking the `kubectl` command but it becomes very complex to pass the correct kubeconfig from java to CLI

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by hypnoce**
## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [x] I Added [CHANGELOG](../CHANGELOG.md) entry regarding this change
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/core/v1/PodOperationsImpl.java [hunk 1]
```diff
@@ -468,33 +470,76 @@ private void copyFile(String source, File target)  {
         String filename = parts[parts.length - 1];
         destination = destination.toPath().resolve(filename).toFile();
     }
-    try (InputStream is = readFile(source);) {
-      Files.copy(is, destination.toPath(), StandardCopyOption.REPLACE_EXISTING);
+
+    try {
+      Files.deleteIfExists(destination.toPath());
```
Comment (shawkins) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/core/v1/PodOperationsImpl.java:475]: Won't the FileOutputStream default to overwriting?

Comment (hypnoce) [reply kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/core/v1/PodOperationsImpl.java]: Right. Fixed.

**Comment by hypnoce** [comment]
> SonarCloud Quality Gate failed.    Quality Gate failed
> 
> [Bug](https://sonarcloud.io/project/issues?id=fabric8io_kubernetes-client&pullRequest=3897&resolved=false&types=BUG) [C](https://sonarcloud.io/project/issues?id=fabric8io_kubernetes-client&pullRequest=3897&resolved=false&types=BUG) [1 Bug](https://sonarcloud.io/project/issues?id=fabric8io_kubernetes-client&pullRequest=3897&resolved=false&types=BUG) [Vulnerability](https://sonarcloud.io/project/issues?id=fabric8io_kubernetes-client&pullRequest=3897&resolved=false&types=VULNERABILITY) [A](https://sonarcloud.io/project/issues?id=fabric8io_kubernetes-client&pullRequest=3897&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=fabric8io_kubernetes-client&pullRequest=3897&resolved=false&types=VULNERABILITY) [Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=fabric8io_kubernetes-client&pullRequest=3897&resolved=false&types=SECURITY_HOTSPOT) [A](https://sonarcloud.io/project/security_hotspots?id=fabric8io_kubernetes-client&pullRequest=3897&resolved=false&types=SECURITY_HOTSPOT) [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=fabric8io_kubernetes-client&pullRequest=3897&resolved=false&types=SECURITY_HOTSPOT) [Code Smell](https://sonarcloud.io/project/issues?id=fabric8io_kubernetes-client&pullRequest=3897&resolved=false&types=CODE_SMELL) [A](https://sonarcloud.io/project/issues?id=fabric8io_kubernetes-client&pullRequest=3897&resolved=false&types=CODE_SMELL) [1 Code Smell](https://sonarcloud.io/project/issues?id=fabric8io_kubernetes-client&pullRequest=3897&resolved=false&types=CODE_SMELL)
> 
> [](https://sonarcloud.io/component_measures?id=fabric8io_kubernetes-client&pullRequest=3897&metric=new_coverage&view=list) [0.0% Coverage](https://sonarcloud.io/component_measures?id=fabric8io_kubernetes-client&pullRequest=3897&metric=new_coverage&view=list) [](https://sonarcloud.io/component_measures?id=fabric8io_kubernetes-client&pullRequest=3897&metric=new_duplicated_lines_density&view=list) [0.0% Duplication](https://sonarcloud.io/component_measures?id=fabric8io_kubernetes-client&pullRequest=3897&metric=new_duplicated_lines_density&view=list)

Not sure about those code smells. One of them is wrong since thread is re-interrupt in another method.

**Review by shawkins — APPROVED** [review]
LGTM.  You can ignore the smell / bug reports when they are not applicable.

**Review by manusa — APPROVED** [review]

**Pull request merged**